From 52ce2a592e4f103535d9d9cecb19aa268ca75c97 Mon Sep 17 00:00:00 2001
From: srihars1 <sriharsha.govinda.raj@intel.com>
Date: Wed, 11 Feb 2015 20:20:31 +0530
Subject: [PATCH] [EM/PMIC]: Close USB phy controller switch only on SDP / CDP

pmic driver needs to open the USB phy switch for DCP.
It should be closed only on SDP / CDP detection.

Change-Id: Ie343bb8120c453af920c892f9310050db3c78f18
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1067
Signed-off-by: Kurtadkar, Yashodhan <yashodhan.kurtadkar@intel.com>
Signed-off-by: srihars1 <sriharsha.govinda.raj@intel.com>
---
 drivers/power/intel_pmic_ccsm.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index 9a2844c..fd78cc7 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -864,23 +864,15 @@ static int get_charger_type(void)
 static void handle_internal_usbphy_notifications(int mask)
 {
 	struct power_supply_cable_props cap = {0};
-	int evt = -1;
+	int evt = USB_EVENT_NONE;
 
 	if (mask) {
 		cap.chrg_evt = POWER_SUPPLY_CHARGER_EVENT_CONNECT;
 		cap.chrg_type = get_charger_type();
 		chc.charger_type = cap.chrg_type;
-		/* Close D+/D- lines in USB detection switch
-		 * due to WC PMIC bug
-		 */
-		pmic_write_reg(chc.reg_map->pmic_usbphyctrl, 0x1);
 	} else {
 		cap.chrg_evt = POWER_SUPPLY_CHARGER_EVENT_DISCONNECT;
 		cap.chrg_type = chc.charger_type;
-		/* Open D+/D- lines in USB detection switch
-		 * due to WC PMIC bug
-		 */
-		pmic_write_reg(chc.reg_map->pmic_usbphyctrl, 0x0);
 	}
 
 	switch (cap.chrg_type) {
@@ -932,6 +924,14 @@ static void handle_internal_usbphy_notifications(int mask)
 	if (cap.chrg_evt == POWER_SUPPLY_CHARGER_EVENT_DISCONNECT)
 		chc.charger_type = POWER_SUPPLY_CHARGER_TYPE_NONE;
 
+	/*
+	 * Open / Close D+/D- lines in USB detection switch
+	 * due to WC PMIC bug only for SDP/CDP.
+	 */
+	pmic_write_reg(chc.reg_map->pmic_usbphyctrl,
+			((evt == USB_EVENT_VBUS)
+				|| (evt == USB_EVENT_ID)) ? 1 : 0);
+
 	atomic_notifier_call_chain(&chc.otg->notifier,
 				USB_EVENT_CHARGER, &cap);
 	if (evt >= 0)
-- 
1.9.1

